event-session-signup/Code.gs

```diff
@@ -0,0 +1,176 @@
+/**
+ * A special function that inserts a custom menu when the spreadsheet opens.
+ */
+function onOpen() {
+  SpreadsheetApp.getUi().createMenu('Conference')
+    .addItem('Set up conference', 'setUpConference_')
+    .addToUi();
+}
+
+/**
+ * A set-up function that uses the conference data in the spreadsheet to create
+ * Google Calendar events, a Google Form, and a trigger that allows the script
+ * to react to form responses.
+ */
+function setUpConference_() {
+  if (ScriptProperties.getProperty('calId')) {
+    Browser.msgBox('Your conference is already set up. Look in Google Drive!');
+  }
+  var ss = SpreadsheetApp.getActive();
+  var sheet = ss.getSheetByName('Conference Setup');
+  var range = sheet.getDataRange();
+  var values = range.getValues();
+  setUpCalendar_(values, range);
+  setUpForm_(ss, values);
+  ScriptApp.newTrigger('onFormSubmit').forSpreadsheet(ss).onFormSubmit()
+      .create();
+  ss.removeMenu('Conference');
+}
+
+/**
+ * Creates a Google Calendar with events for each conference session in the
+ * spreadsheet, then writes the event IDs to the spreadsheet for future use.
+ * @param {Array<string[]>} values Cell values for the spreadsheet range.
+ * @param {Range} range A spreadsheet range that contains conference data.
+ */
+function setUpCalendar_(values, range) {
+  var cal = CalendarApp.createCalendar('Conference Calendar');
+  for (var i = 1; i < values.length; i++) {
+    var session = values[i];
+    var title = session[0];
+    var start = joinDateAndTime_(session[1], session[2]);
```

Comment:
Since no timezones referenced anywhere, would help to clarify what the expectations are for the times. Is it using the sheet/script timezone? User's timezone? Any oddities depending on who runs setup?

Doesn't necessarily mean there are bugs, just need to be clear about what the times represent.

Reply:
Ooooh this is a good point, so _setUpForm (~94) uses:
    var day = session[1].toLocaleDateString();
    var time = session[2].toLocaleTimeString();

so I see EST in the form:
<img width="588" alt="Screen Shot 2019-06-07 at 5 48 52 PM" src="https://user-images.githubusercontent.com/380123/59135707-efca2400-894d-11e9-9718-59e32a6df3a0.png">
 
I'll add a comment that it is the uses the 'conference organizer's' timezone